### PR TITLE
Garden create endpoint

### LINKED ISSUE #35: Garden Create Endpoint
A user can send a request to `POST /api/v1/gardens`, with data for creating a garden. It successfully creates and returns the garden.  The user is identified and authenticated by their api key sent in the header of the request.

Request Format:
```
POST /api/v1/gardens
Content-Type: application/json
Accept: application/json
Authorization: <user_api_key>

 {
        "garden": {
            "name": "Backyard Raised Bed",
            "latitude": 39.742043,
            "longitude": -104.991531,
            "max_moisture": 82.5,
            "min_moisture": 22.5,
            "auto_water": "false"
        }
 }
```

Response Format:
```
status: 201
body:
{
    "data": {
        "id": "1",
        "type": "garden",
        "attributes": {
            "id": 1,
            "name": "Backyard Raised Bed",
            "latitude": 39.742043,
            "longitude": -104.991531,
            "max_moisture": 82.5,
            "min_moisture": 22.5,
            "auto_water": "false"
        }
    }
}
```

### Checks
- [x] For success, returns status code 201
- [x] Returns correct response format json

**Sad path**
- [x] Error handling for unauthorized user and invalid garden parameters

### PR DESCRIPTION
**Pull request by Mycobee**
## Changes proposed in this pull request:
* This PR adds the create specs and functionality to the garden model.  Validations are added for presence of name, latitude, and longitude.   Also added are numericality validations for latitude, longitude, max_moisture, and min_moisture. A conditional presence validation for min_moisture is added when `Garden.auto_water` is true.   A migration adds a default value of false to auto_water.  Error handling is present for both bad requests and invalid credentials. 

## Card(s) closed in this pull request:
close #35

## What did you struggle on to complete?
* No current blockers

### Model Test Coverage: 100%
- [x] Tested my new feature(s) as well as any feasible edge cases (if possible)
- [x] Checked coverage/index.html - did not add any new code that's not covered by testing (if possible)
- [x] Ran the test suite - all tests are passing
- [x] Merged in the latest master to my branch with git pull origin master & resolved merge conflicts
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: app/controllers/api/v1/gardens_controller.rb [hunk 1]
```diff
@@ -1,14 +1,34 @@
 class Api::V1::GardensController < ApplicationController
   before_action :set_garden, only: [:show]
+  before_action :authenticate_user, only: [:create]
```
Comment (bplantico) [inline app/controllers/api/v1/gardens_controller.rb]: 👍

File: app/controllers/api/v1/gardens/jobs_controller.rb [hunk 2]
```diff
@@ -29,6 +29,6 @@ def job_params
 
   def set_garden
     @garden = Garden.find_by(id: params[:id])
-    not_found if @garden.nil?
+    garden_not_found if @garden.nil?
```
Comment (bplantico) [inline app/controllers/api/v1/gardens/jobs_controller.rb]: Very explicit, nice change.

File: spec/requests/api/v1/gardens/gardens_request_spec.rb [hunk 3]
```diff
@@ -24,4 +26,72 @@
...
+    expect(error).to eq("Invalid Credentials")
+	end
+
+  it "returns 400 for a bad request" do
+    headers = { "Authorization": @user.api_key } 
+
+    post "/api/v1/gardens", params: {
+      "garden": {
+        "name": "Backyard Raised Bed",
+        "latitude": "Chicken Nuggets",
```
Comment (bplantico) [inline spec/requests/api/v1/gardens/gardens_request_spec.rb]: These sound really good right now.

File: app/models/garden.rb [hunk 4]
```diff
@@ -1,5 +1,7 @@
 class Garden < ApplicationRecord
 	# validations
+  validates_presence_of :name, :latitude, :longitude
+  validates_numericality_of :latitude, :longitude, :max_moisture, :min_moisture
```
Comment (bplantico) [inline app/models/garden.rb]: @Mycobee Thoughts on adding conditional validation that min_moisture has to be present if auto_water is set to true?

Comment (Mycobee) [reply app/models/garden.rb]: Done in the new changes 👍

File: app/controllers/api/v1/gardens_controller.rb (hunk 1 above)
Comment (Mycobee) [reply app/controllers/api/v1/gardens_controller.rb]: I figure this method will be useful for us over the next few days.

File: app/models/garden.rb [hunk 5]
```diff
@@ -2,6 +2,7 @@ class Garden < ApplicationRecord
 	# validations
   validates_presence_of :name, :latitude, :longitude
   validates_numericality_of :latitude, :longitude, :max_moisture, :min_moisture
+  validates :min_moisture, presence: true, if: :auto_water
```
Comment (bplantico) [inline app/models/garden.rb]: SWEET!

File: spec/models/garden_spec.rb [hunk 6]
```diff
@@ -9,6 +9,11 @@
     it { should validate_numericality_of :longitude  }
     it { should validate_numericality_of :max_moisture  }
     it { should validate_numericality_of :min_moisture  }
+
+    context "if auto_water" do
```
Comment (bplantico) [inline spec/models/garden_spec.rb]: Awesome @Mycobee Hopefully this wasn't too tough to figure out testing?
